File: concurrency_tools_practice/app/src/main/java/com/test/concurrency_tools_practice/MainActivity.java
package com.test.concurrency_tools_practice;

import androidx.appcompat.app.AppCompatActivity;

import android.os.AsyncTask;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        AsyncTask<Object, Object, Object> execute = new AsyncTask() {
            @Override
            protected Object doInBackground(Object[] objects) {
                return null;
            }
        }.execute();
    }
}

File: kotlin_coroutine_learn/app/src/main/java/com/test/kotlin_coroutine/MainActivity.kt
package com.test.kotlin_coroutine

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ImageView
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

class MainActivity : AppCompatActivity() {

    private var image1: ImageView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        image1 = findViewById(R.id.image1)

        CoroutineScope(Dispatchers.Main).launch {
            var bitmap = withContext(Dispatchers.IO) {
                //异步获取图片后，设置到image1上
            }
//            image1?.setImageBitmap(bitmap)


        }

    }
}